lib/src/ast.dart

```diff
@@ -9,15 +20,22 @@ abstract class Query {
   /// testing unambiguous.
   @override
   String toString({bool debug = false});
+
+  R cast<R extends Query>() => this as R;
```

Comment:
Please add a line of documentation, something could be copy-pasted from https://api.dart.dev/stable/2.17.6/dart-core/List/cast.html